src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -1074,6 +1075,7 @@ def _read_semi_structured_file(self, path: str, format: str) -> DataFrame:
         set_api_call_source(df, f"DataFrameReader.{format.lower()}")
         return df
 
+    @private_preview(version="1.29.0")
     @publicapi
```

Comment:
is @publicapi conflict with @private_preview?

Reply:
as far as I can tell, publicapi is for sending ast while private_preview serves as feature flag
it indeed looks confusing to users. I can remove the publicapi for now